[FEAT] 게시글- 게시글API

boojang opened the pull request:
### Description

게시글 진행상황 업로드
- 게시글 리스트 조회
- 게시글 등록
- 게시글 주소 관리
- 게시글 이미지 관리

### Related Issues

#11 
#12

boojang committed:
feat: 게시글 DTO 추가

- PostCreateRequestDto 클래스 추가: 게시글 생성 요청에 필요한 데이터 필드 정의
- PostCreateResponseDto 클래스 추가: 게시글 생성 응답에 필요한 데이터 필드 정의
- PostDetailResponseDto 클래스 추가: 게시글 상세 정보 응답에 필요한 데이터 필드 정의
- PostListResponseDto 클래스 추가: 게시글 목록 응답에 필요한 데이터 필드 정의
- PostRequestDto 클래스 삭제: 더 이상 사용되지 않는 클래스를 제거하여 코드 정리.

boojang committed:
feat: PostController, PostService 수정정

- PostController: 게시글 목록 조회 및 게시글 등록 기능을 제공하는 REST API 엔드포인트 구현
- PostService: 게시글 목록 조회 및 게시글 등록 로직을 포함하여 비즈니스 로직을 처리
- 게시글 등록 시 이미지 및 태그 처리 로직 추가
- 요청 데이터 유효성 검증 및 예외 처리 로직 구현

boojang committed:
feat: 게시글 - 주소 관련 기능 추가

- AddressController : 주소 관련 요청을 처리하는 컨트롤러 구현
- AddressDto : 주소 정보를 담기 위한 데이터 전송 객체 정의
- AddressRepository : 주소 데이터에 대한 CRUD 작업을 위한 JPA 리포지토리 정의
- AddressService : 주소 정보를 저장하거나 조회하는 비즈니스 로직 구현
- 주소 정보 유효성 검증 및 예외 처리 로직 추가

boojang committed:
feat: 게시글, 태그 리포지토리 추가

- PostRepository: 게시글 관련 데이터 접근을 위한 JPA 리포지토리 구현
  - 활성 게시글 목록 조회 및 주소 정보 포함 조회 기능 추가
  - 특정 게시글 ID로 활성 게시글 조회 기능 추가
- TagRepository: 태그 관련 데이터 접근을 위한 JPA 리포지토리 구현
  - 태그 이름으로 태그 조회 및 이름 포함 검색 기능 추가

boojang committed:
feat: S3 이미지 업로드 및 삭제 기능 추가

- S3Config: AWS S3 클라이언트를 설정하는 구성 클래스 추가
- S3Service: S3에 이미지를 업로드하고 삭제하는 기능을 정의하는 인터페이스 추가
- S3ServiceImpl: S3Service 인터페이스를 구현하여 이미지 업로드 및 삭제 로직을 포함하는 서비스 클래스 추가
  - MultipartFile 형식의 이미지를 S3에 업로드하고 URL 반환
  - 다수의 이미지를 S3에 업로드하고 URL 목록 반환
  - S3에서 이미지 삭제 기능 구현
  - 파일명에서 확장자 추출 및 S3 URL에서 키 추출 기능 추가
  - 예외 처리 및 로깅 기능 포함

boojang committed:
feat: 오류 코드 추가 및 수정

- ErrorCode 열거형에 새로운 오류 코드 추가:
  - USER_NOT_FOUND 코드의 키를 "not_found_user"로 수정
  - INVALID_TAG_NAME 코드 추가: 존재하지 않는 태그에 대한 오류 처리
  - FAILED_TO_UPLOAD_IMAGE 코드 추가: 이미지 업로드 실패에 대한 오류 처리

ghkddlscks19 reviewed (changes requested):
address 확인하는 로직 제거
responsedto 유효성 관련 어노테이션 제거
NotNull NotBlank 알아보고 적용
유효성 검사할 때 Controller단에 @Valid 붙여주기

boojang committed:
feat: 게시글 상세 조회 및 수정, 삭제 기능 추가

- PostController: 게시글 상세 조회, 수정 및 삭제를 위한 새로운 API 엔드포인트 추가
  - getPostDetail: 게시글 ID로 상세 정보를 조회하는 메서드 구현
  - updatePost: 게시글 수정 기능 추가, 주소 및 이미지 업데이트 포함
  - deletePost: 게시글 삭제 기능 추가, 소프트 딜리트 구현
- PostService: 게시글 상세 조회 및 수정 로직 추가
  - getPostDetail: 게시글 상세 정보를 DTO로 반환하는 메서드 구현
  - updatePost: 게시글 수정 시 주소 및 태그 업데이트 로직 추가
- PostCreateRequestDto: 게시글 제목 및 내용 필드에 @NotBlank 유효성 검증 추가
- ErrorCode: 게시글 수정 실패에 대한 새로운 오류 코드 추가

boojang committed:
Merge branch 'develop' of https://github.com/100-hours-a-week/4-team-4dollar-matajo-be into feat/board

boojang committed:
feat: 주소 정보 업데이트 기능 추가

- Address 클래스에 주소 정보를 업데이트하는 update 메서드 추가
- AddressService 클래스에 주소 정보를 업데이트하는 updateAddress 메서드 추가
  - 유효성 검증 로직 포함: 주소 정보가 null이거나 비어있을 경우 예외 발생
  - 주소 정보 업데이트 후 로그 기록

boojang committed:
feat: 게시글 수정 및 공개 상태 전환 기능 추가

- PostController: 게시글 수정 API의 응답 DTO를 PostCreateResponseDto로 변경하고, 게시글 공개 상태를 전환하는 togglePostVisibility 메서드 추가
- PostService: 게시글 수정 시 주소 및 이미지 업데이트 로직 개선, 게시글 공개/비공개 상태 전환 메서드 구현
- Post: 게시글 공개 상태 전환을 위한 toggleHiddenStatus 메서드 추가 및 주소 업데이트 메서드 개선
- PostRepository: 주소 정보 조회 쿼리 복원

boojang committed:
feat: 보안 및 CORS 설정 추가

- SecurityConfig: Spring Security를 사용하여 CSRF 보호를 비활성화하고 모든 요청에 대해 인증 없이 접근을 허용하는 보안 필터 체인 추가
- WebConfig: CORS 설정을 추가하여 프론트엔드 서버 주소(http://localhost:3000)에 대한 요청을 허용하고, 다양한 HTTP 메서드 및 헤더에 대한 접근을 설정
- UserRepository: 사용자 엔티티에 대한 JPA 리포지토리를 구현하고 Kakao ID로 사용자를 조회하는 메서드 추가
